tests/test_edgeql_expressions.py

```diff
@@ -4175,6 +4175,22 @@ async def test_edgeql_expr_setop_11(self):
             len(everything), len(distinct),
             'DISTINCT len(ObjectType.name) failed to filter out dupplicates')
 
+    async def test_edgeql_expr_setop_12(self):
+        await self.assert_query_result(
+            r'''SELECT DISTINCT (SELECT ({1,2,3}, ()) FILTER .0 > 1).1;''',
```

Comment:
Let's also add the explicit set literal case `SELECT DISTINCT {(), ()}` just in case set literals get some special handling.